classes/task/PKPUsageStatsLoader.php

```diff
@@ -114,88 +105,31 @@ protected function executeActions(): bool
-    protected function deleteByLoadId(string $loadId): void
-    {
-        $this->temporaryInstitutionsDao->deleteByLoadId($loadId);
-        $this->temporaryTotalsDao->deleteByLoadId($loadId);
-        $this->temporaryItemInvestigationsDao->deleteByLoadId($loadId);
-        $this->temporaryItemRequestsDao->deleteByLoadId($loadId);
-    }
 
-    /**
-     * Insert usage stats log entry into temporary tables
-     */
-    protected function insertTemporaryUsageStatsData(object $entry, int $lineNumber, string $loadId): void
-    {
-        try {
-            $this->temporaryTotalsDao->insert($entry, $lineNumber, $loadId);
-            $this->temporaryInstitutionsDao->insert($entry->institutionIds, $lineNumber, $loadId);
-            if (!empty($entry->submissionId)) {
-                $this->temporaryItemInvestigationsDao->insert($entry, $lineNumber, $loadId);
-                if ($entry->assocType == Application::ASSOC_TYPE_SUBMISSION_FILE) {
-                    $this->temporaryItemRequestsDao->insert($entry, $lineNumber, $loadId);
-                }
+        if ($processFilesResult) {
+            $site = Application::get()->getRequest()->getSite();
+            $jobs = [];
+            foreach ($this->logFiles as $filePath) {
+                $jobsPerFile = $this->getFileJobs($filePath, $site);
+                $jobs = array_merge($jobs, $jobsPerFile);
             }
-        } catch (\Illuminate\Database\QueryException $e) {
-            $this->addExecutionLogEntry(__('admin.scheduledTask.usageStatsLoader.insertError', ['file' => $loadId, 'lineNumber' => $lineNumber, 'msg' => $e->getMessage()]), ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR);
-        }
-    }
-
-    /**
-     * Get valid assoc types that an usage event can contain
-     */
-    abstract protected function getValidAssocTypes(): array;
-
-    /**
-     * Validate the usage stats log entry
-     *
-     * @throws Exception.
-     */
-    protected function isLogEntryValid(object $entry): void
-    {
-        if (!$this->validateDate($entry->time)) {
-            throw new Exception(__('admin.scheduledTask.usageStatsLoader.invalidLogEntry.time'));
-        }
-        // check hashed IP ?
-        // check canonicalUrl ?
-        if (!is_int($entry->contextId)) {
-            throw new Exception(__('admin.scheduledTask.usageStatsLoader.invalidLogEntry.contextId'));
-        }
-        if (!empty($entry->submissionId) && !is_int($entry->submissionId)) {
-            throw new Exception(__('admin.scheduledTask.usageStatsLoader.invalidLogEntry.submissionId'));
-        }
+            foreach ($this->months as $month) {
+                $compileMonthlyMetricsJob = new CompileMonthlyMetrics($month, $site);
+                $jobs = array_merge($jobs, [$compileMonthlyMetricsJob]);
+            }
+            // Bus::chain() cannot accept an empty array
+            if (!empty($jobs)) {
+                Bus::chain($jobs)
+                    ->catch(function (Throwable $e) {
```

Comment:
What types of errors can come here? If there's something useful, it might be better to log, the `Throwable` can catch even code errors.

Reply:
> When chaining jobs, you may use the catch method to specify a closure that should be invoked if a job within the chain fails. The given callback will receive the Throwable instance that caused the job failure

OK, yes, I will log the error...

Reply:
Hmmm... I think the error message is then visible in the failed_jobs table -- that's why I haven't done anything here...

Maybe to ask @touhidurabir if I should also log the error here?

Reply:
we don't need to log the anything if we only need to log the exception as those do get logged in failed_jobs and also in the error log file as we have the `JobFailed` event register trough `Queue::failing`. We can use that `catch` if we need to run anything extra if a chain job failed. more details at https://laravel.com/docs/10.x/queues#chain-failures .